ui/static/ui/js/learning_resources.jsx

```diff
@@ -120,8 +127,65 @@ define('learning_resources', [
         _.filter(e.target.options, function(option) {
           return option.selected && option.value !== null;
         }), 'value');
-
-      this.props.setValues(this.props.selectedVocabulary.slug, selectedValues);
+      // check if the current vocabulary allows free tagging and in case add
+      // the new tags before proceeding
+      if (this.props.selectedVocabulary.vocabulary_type === 'f') {
+        // extract the list of terms not in the vocabulary existing terms
+        var termsSlugs = _.pluck(this.props.selectedVocabulary.terms, 'slug');
+        var termsToCreate = _.filter(selectedValues, function (slug) {
+          return _.indexOf(termsSlugs, slug) === -1;
+        });
+        // if there are no new terms to create, set the state
+        if (!termsToCreate.length) {
+          this.props.setValues(
+            this.props.selectedVocabulary.slug, selectedValues
+          );
+        } else {
+          // otherwise create the terms and update the vocabularies
+          // create the term
+          var thiz = this;
+          var API_ROOT_TERMS_URL = '/api/v1/repositories/' +
+            thiz.props.repoSlug + '/vocabularies/' +
+            thiz.props.selectedVocabulary.slug + "/terms/";
+          _.forEach(termsToCreate, function (termLabel) {
+            $.ajax({
+                type: "POST",
+                url: API_ROOT_TERMS_URL ,
+                data: JSON.stringify({
+                  label: termLabel,
+                  weight: 1
+                }),
+                contentType: "application/json; charset=utf-8"
+              }
+            )
+            .fail(function() {
+                thiz.props.reportMessage({
+                  error: "Error occurred while adding new term \"" +
+                    termLabel + "\"."
+                });
+              })
+            .done(function(newTerm) {
+              // replace che current label in the selected
+              // values with the newly created slug
+              selectedValues.splice(
+                _.indexOf(selectedValues, termLabel),
+                1,
+                newTerm.slug
+              );
+              // add the term to the current selected vocabulary
+              thiz.props.selectedVocabulary.terms.push(newTerm);
```

Comment:
We should not alter `props`, `props` are supposed to be immutable. Either use a callback to have the parent make this change in the props, or move this functionality into the state if necessary.